src/plugins/query_enhancements/public/search/sql_async_search_interceptor.ts

```diff
@@ -71,33 +78,43 @@ export class SQLAsyncSearchInterceptor extends SearchInterceptor {
       ...dataFrame.meta,
       queryConfig: {
         ...dataFrame.meta.queryConfig,
-        ...(this.connectionsService.getSelectedConnection() &&
-          this.connectionsService.getSelectedConnection()?.dataSource && {
-            dataSourceId: this.connectionsService.getSelectedConnection()?.dataSource.id,
-          }),
+        ...(this.connectionsService.getSelectedConnection() && {
+          dataSourceId: this.connectionsService.getSelectedConnection()?.id,
+        }),
       },
     };
+    const queryId = uuid();
+    // Send an initial submit event to get faster feedback to clients waiting for info, since
+    // polling will wait for the first polling cycle to finish before sending anything
+    this.uiActions?.getTrigger(ASYNC_TRIGGER_ID).exec({
+      queryId,
+      queryStatus: JobState.SUBMITTED,
+    });
 
     const onPollingSuccess = (pollingResult: any) => {
-      if (pollingResult && pollingResult.body.meta.status === 'SUCCESS') {
-        return false;
-      }
-      if (pollingResult && pollingResult.body.meta.status === 'FAILED') {
-        const jsError = new Error(pollingResult.data.error.response);
-        this.deps.toasts.addError(jsError, {
-          title: i18n.translate('queryEnhancements.sqlQueryError', {
-            defaultMessage: 'Could not complete the SQL async query',
-          }),
-          toastMessage: pollingResult.data.error.response,
+      if (pollingResult) {
+        const queryStatus = parseJobState(pollingResult.body.meta.status)!;
```

Comment:
I don't think we should trust that we will always get a valid `body.meta.status`.

Reply:
Refactored to guard against null: if an invalid or missing status is returned we'll report an error and stop polling.